02-peer-protocol.md

```diff
@@ -49,11 +68,468 @@ transaction is confirmed are also not persistent - until you know the script
+    - the `feerate` is not greater than 25/24 times `feerate` of the last
+      successfully constructed transaction
+  - MAY fail the negotiation for any reason
+
+#### Rationale
+
+`feerate` is the feerate this transaction will pay. It must be at least
+1/24 greater than the last used `feerate`, rounded down to the nearest
+satoshi to ensure there is progress.
+
+E.g. if the last `feerate` was 520, the next sent `feerate` must be 541
+(520 * 25 / 24 = 541.667, rounded down to 541).
+
+If the previous transaction confirms in the middle of an RBF attempt,
+the attempt MUST be abandoned.
+
+`funding_output_contribution` is the amount of satoshis that this peer
+will contribute to the funding output of the transaction, when there is
+such an output. Note that it may be different from the contribution
+made in the previously completed transaction.
+
+### The `tx_ack_rbf` Message
+
+1. type: 73 (`tx_ack_rbf`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`tx_ack_rbf_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `tx_ack_rbf_tlvs`
+2. types:
+    1. type: 0 (`funding_output_contribution`)
+    2. data:
+        * [`tu64`:`satoshis`]
+
+#### Requirements
+
+The recipient:
+  - MUST either fail the negotiation or transmit a `tx_add_input` message,
+    restarting the interactive tx collaboration protocol.
+
+#### Rationale
+
+`funding_output_contribution` is the amount of satoshis that this peer
+will contribute to the funding output of the transaction, when there is
+such an output. Note that it may be different from the contribution
+made in the previously completed transaction.
+
+It's recommended that a peer, rather than fail the RBF negotiation due to
+a large feerate change, instead sets their `funding_output_contribution` to
+zero, and decline to participate further in the transaction.
+
+### The `tx_abort` Message
+
+1. type: 74 (`tx_abort`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u16`:`len`]
+   * [`len*byte`:`data`]
+
+The `len` field indicates the number of bytes in the immediately following field.
+
+#### Requirements
+
+A sending node:
+  - MUST NOT have already transmitted `tx_signatures`
+  - SHOULD forget the current negotiation and reset their state.
+
+A receiving node:
+  - if they have already sent `tx_signatures` to the peer:
+    - MUST NOT forget the channel until any inputs to the negotiated tx
+      have been spent.
+  - if they have not sent `tx_signatures`:
+    - SHOULD forget the current negotiation and reset their state.
+
+#### Rationale
+
+A receiving node, if they've already sent their `tx_signatures` has no guarantee
+that the transaction won't be signed and published by their peer. They must remember
+the transaction and channel (if appropriate) until the transaction is no longer
+eligible to be spent (i.e. any input has been spent in a different transaction).
+
+The `tx_abort` message allows for the cancellation of an in progress negotiation,
+and a return to the initial starting state.
```

Comment:
Because you don't want to fail the channel when you are in an `interactive-tx` session to RBF the funding tx for example. In that case you just want to fail the RBF `interactive-tx` session, but the previous funding txs may confirm so you don't want to fail the channel.

Reply:
Makes sense.  We should explain this under Rationale, and probably define the term "fail the negotiation" (as opposed to "fail the channel") in the Glossary.

Do we always send `tx_abort` instead of `error` in v2?  Only during the `interactive-tx` part?  Only during RBF?

Do we send `tx_abort` for some errors, and `error` for others?

Reply:
> Do we always send tx_abort instead of error in v2? Only during the interactive-tx part? Only during RBF?
> Do we send tx_abort for some errors, and error for others?

You should send `tx_abort` when you want to abort an `interactive-tx` protocol, but don't necessarily want to reject the channel.
You should send `error` when you want to reject the whole channel opening flow.

Those are really two different intents, and you'll see in the implementation that it's somewhat obvious which one you should use depending on what internal error is hit.